stan/math/prim/fun/cholesky_corr_free.hpp

```diff
@@ -10,24 +10,25 @@
 namespace stan {
 namespace math {
 
-template <typename T>
-Eigen::Matrix<T, Eigen::Dynamic, 1> cholesky_corr_free(
-    const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& x) {
+template <typename T, require_eigen_t<T>* = nullptr>
+Eigen::Matrix<value_type_t<T>, Eigen::Dynamic, 1> cholesky_corr_free(
+    const T& x) {
   using Eigen::Dynamic;
   using Eigen::Matrix;
 
   check_square("cholesky_corr_free", "x", x);
   // should validate lower-triangular, unit lengths
 
+  const Eigen::Ref<const plain_type_t<T>> x_ref = x;
   int K = (x.rows() * (x.rows() - 1)) / 2;
-  Matrix<T, Dynamic, 1> z(K);
+  Matrix<value_type_t<T>, Dynamic, 1> z(K);
   int k = 0;
   for (int i = 1; i < x.rows(); ++i) {
-    z(k++) = corr_free(x(i, 0));
-    double sum_sqs = square(x(i, 0));
+    z(k++) = corr_free(x_ref(i, 0));
+    double sum_sqs = square(x_ref(i, 0));
```

Comment:
Should these all be `coeffRef`?

Reply:
`coeffRef` is required when writing values. For reading I think `coeff` is prefered.